### PR TITLE
m2 mainnet / goerli v2 contracts

### PR DESCRIPTION
**Pull request by shrimalmadhur**
Fixes # .

- Updated to change Registration process. Now this only registered with Delegation manager. BLS key registration happens during AVS registration
- Added a migration of config since new methods needs Delegation address instead of Slasher address.
- Removing BLS key path from operator registration to EigenLayer process as it is not needed anymore for EigenLayer registration. We do need BLS key for EigenDA registration

TODO:

- [ ]  test against actual new contracts
- [ ] PR for eigenlayer docs website

### CODE REVIEW

File: pkg/operator/register.go [hunk 1]
```diff
@@ -13,14 +16,32 @@ import (
...
+}
+
+var chainMetadataMap = map[int64]ChainMetadata{
+	utils.MainnetChainId: {
+		BlockExplorerUrl:           "https://etherscan.io/tx",
+		ELDelegationManagerAddress: "",
+	},
+	utils.GoerliChainId: {
+		BlockExplorerUrl:           "https://goerli.etherscan.io/tx",
+		ELDelegationManagerAddress: "0x1b7b8F6b258f95Cf9596EabB9aa18B62940Eb0a8",
```
Comment (shrimalmadhur) [inline pkg/operator/register.go:37]: @ypatil12 can you confirm this is right?

Comment (ypatil12) [reply pkg/operator/register.go:37]: Look good

File: pkg/operator/register.go [hunk 2]
```diff
@@ -13,14 +16,32 @@ import (
...
+	},
+	utils.HoleskyChainId: {
+		BlockExplorerUrl:           "https://holesky.etherscan.io/tx",
+		ELDelegationManagerAddress: "",
+	},
+}
+
 func RegisterCmd(p utils.Prompter) *cli.Command {
 	registerCmd := &cli.Command{
 		Name:      "register",
```
Comment (ypatil12) [inline pkg/operator/register.go:47]: Comment below: `Register the operator and the BLS public key in the EigenLayer contracts` is stale?

File: pkg/operator/config/create.go [hunk 3]
```diff
@@ -66,7 +66,7 @@ func CreateCmd(p utils.Prompter) *cli.Command {
 			}
 
 			fmt.Println(
-				"Created operator.yaml and metadata.json files. Please fill in the smart contract configuration details(el_slasher_address and bls_public_key_compendium_address) provided by EigenLayer team. ",
+				"Created operator.yaml and metadata.json files. Please fill in the smart contract configuration details(el_delegation_manager_address) provided by EigenLayer team. ",
```
Comment (samlaf) [inline pkg/operator/config/create.go:69]: can't we hardcode this address?

**Review by samlaf — APPROVED** [review]
LGTM

File: pkg/operator/register.go (hunk 2 above)
Comment (shrimalmadhur) [reply pkg/operator/register.go:47]: good catch

Comment (shrimalmadhur) [reply pkg/operator/register.go:47]: fixed

**Review by ypatil12 — APPROVED** [review]

File: pkg/operator/config/create.go (hunk 3 above)
Comment (shrimalmadhur) [reply pkg/operator/config/create.go]: just added.

**Pull request merged**